### ISSUE TITLE
global pip (`--system` ...) doesn't work in GitHub Codespaces

### ISSUE THREAD

**Issue by Malix-off** [astral-sh/uv#3417]
## Bug

global pip (`--system` ...) doesn't work in GitHub Codespaces

## System

- VSCode:
	Version: 1.89.0
	Commit: b58957e67ee1e712cebf466b995adf4c5307b2bd
	Date: 2024-05-01T02:15:54.732Z
	Browser: Mozilla/5.0 (X11; CrOS x86_64 14541.0.0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/125.0.0.0 Safari/537.36
- uv 0.1.39

## Reproduce

1. Open a GitHub Codespace
	- `.devcontainer/devconainer.json`: 
		```json
		{
			"name": "Python 3 on Debian 12",
			"image": "mcr.microsoft.com/devcontainers/python:1-3-bookworm",
			"features": {
				"ghcr.io/va-h/devcontainers-features/uv:1": {}
		}
		```
2. Terminal input: `uv pip install prettytable --system`

## Logs

```log
@Malix-off ➜ /workspaces/Fork (master) $ uv pip install prettytable --system
Resolved 2 packages in 2ms
error: Failed to install: prettytable-3.10.0-py3-none-any.http.whl (prettytable==3.10.0)
  Caused by: failed to create directory `/usr/local/lib/python3.12/site-packages/prettytable-3.10.0.dist-info`
  Caused by: Permission denied (os error 13)
```
_Note: `prettytable` is just an exemple_

**Comment by zanieb** [comment]
Hi! It looks you don't have the necessary permissions to write to the system packages?

**Comment by charliermarsh** [comment]
Probably works in pip because pip falls back to installing with `--user` if the site-packages directory isn't writeable.

**Comment by Malix-off** [comment]
> Hi! It looks you don't have the necessary permissions to write to the system packages?

Indeed, probably like that in GitHub Codespaces, you're right.

> --user

Is that an equivalent for sync?
```log
@Malix-off ➜ /workspaces/Fork (master) $ uv pip sync requirements.txt --user
error: pip-sync's `--user` is unsupported.
```

**Comment by charliermarsh** [comment]
No, we don't support `--user` right now, since it can generally be replaced by a virtual environment. There are some open issues tracking it but we haven't committed to adding it.

**Comment by Malix-off** [comment]
So what would be the workaround to `pip sync` without virtual environment (devcontainer is a good example)

**Comment by charliermarsh** [comment]
I think you basically _need_ to use a virtualenv in that case with uv right now. I believe you _can_ use virtual environments in devcontainers? Like it is possible, right?

**Comment by Malix-off** [comment]
> I think you basically _need_ to use a virtualenv in that case with uv right now. I believe you _can_ use virtual environments in devcontainers? Like it is possible, right?

Yep, currently you need a venv to do that, and yep, it is definitely possible, but adds an unnecessary virtual environment bloat (since we're already in a containerized environment)

**Comment by Malix-off** [comment]
> No, we don't support `--user` right now, since it can generally be replaced by a virtual environment. There are some open issues tracking it but we haven't committed to adding it.

I didn't find any issue about it containing "user" "-u" "--user", do you know if it has been deleted?

**Comment by Malix-off** [comment]
GitHub Codespaces probably doesn't let you access to the devcontainer system (immutability)

In the case of devcontainers, you probably indeed would want to use the `--user` flag (#2077)
